Chore/update integration example (#254)
nmshd/documentation: _docs_integrate/integration-example.md

}
```

You can query the Connector's address via the route `/api/v2/Account/IdentityInfo`. If you are using the Demo Connector of this Tutorial, the address is `did:e:pilot.enmeshed.eu:dids:9364c879fcc41c87c6ecbd`.
{: .notice--info}

> 

Reviewer: why did you remove this? This is still perfectly valid.

Author: The Connector address is no longer required for creating an attribute, so why should it be mentioned in the tutorial? Limit the example to the most necessary to minimize complexity.

Reviewer: ![image](https://github.com/user-attachments/assets/3d15ce65-c972-402b-ba40-b7e4a8628bd3)

?

Reviewer: @britsta is the right side in the screenshot even correct?

Reviewer: also - do we really don't need the address in the whole page?

Author: before you only needed it when creating the attribute, since the update I need it in your screenshot above, if that's what you want and not a bug (it can no longer be “”)

Reviewer: but you need it, so you should re-add the information on how to get to the information..

Author: in line 82: the response from creating the attribute gives you the address

Reviewer: but how should I get the connection that <your-connector-address> is the owner you copied before... makes no sense for me.

Author: why do I now need the owner for the request and can no longer “”, is that intentional? 
And I get the id and the owner when creating the attribute, why can't I use both of them.

Author: I get: "error.consumption.requests.invalidRequestItem: 'The provided IdentityAttribute belongs to someone else. You can only share own IdentityAttributes.' at 2024-08-23T14:00:11.876Z",
I think this is a bug. If the bug is fixed the owner can be “” and then the owner of the IdentityInfo is not needed anywhere

Reviewer: @britsta wdym?

Reviewer 2: It is intended that an empty string can no longer be specified here as the owner, for several reasons:
- The empty string is often used as a placeholder for other RequestItems if the recipient of the Request is not yet known (i.e. if the Request is sent via a RelationshipTemplate and not a Message). To avoid confusion, the empty string should mainly be used for this purpose and no longer as a way of saving typing work for an already know address (e.g. my address).
- With a ShareAttributeRequestItem, we can share an IdentityAttribute that already exists in the database, in which the address of the `owner` is explicitly specified. Normally, it should be copied from the database and can then be inserted in the `attribute` property of the ShareAttributeRequestItem in exactly this form (with specified `owner` and not empty string).
- In the past, it was possible with some RequestItems, if you wanted to cause a bug, that Attributes could be stored in the database that actually had an empty string as `owner`. This should hopefully be fixed by now, but has led us to become stricter here and only want to use an empty string if we have no choice, i.e. do not know an address.

Author: Then the error message I get when validating is incorrect or at least misleading.
But I still see no reason to use: `/api/v2/Account/IdentityInfo`

Reviewer 2: I would actually have expected a different error message as well and also found the problem in the code, will be fixed soon.

Reviewer 2: The PR https://github.com/nmshd/runtime/pull/250 has just been merged so that an error message is now displayed that makes more sense.

When creating an IdentityAttribute, the address of the `owner` can be found in the success response, so that you do not necessarily have to use the use case to query the currently used Identity to get the address. I would probably mention it in a `{: .notice--info}` box anyway, because I think it would be beneficial for the reader.

Author: I am very dissatisfied, if it says save the id and the user from the response and in the note directly below it says oh if you forgot to save the user you can also do that with the use-case.

Reviewer: true, but if you should have copy pasted this why is there a placeholder?

> {% include copy-notice description="Save the `id` and the `owner` of the Attribute that you can find in the Response. You will need it in the next step." %}

Author: @Magnus-Kuhn response always in lowercase? is often capitalized in the file

> {% include copy-notice description="Save the `id` and the `owner` of the Attribute that you can find in the Response. You will need it in the next step." %}

Reviewer: it depends, if we mean a http response then small, if we mean an enmeshed Response, then capital

            "mustBeAccepted": true,
            "attribute": {
              "@type": "IdentityAttribute",

>               "owner": "<the owner of the Attribute created above>",

Reviewer: ```
              "owner": "<the `owner` of the Attribute created above>",
```
or
```
              "owner": "<the owner propery of the Attribute created above>",
```
I like the second more :)

Author: you mean property?

Reviewer: yep

### Connector: Create a RelationshipTemplate

> If the response successfully confirms all attributes, we can create the RelationshipTemplate.

Reviewer: ```suggestion
If the response is successful, we can create the RelationshipTemplate.
```

it sounds like this would only validate attributes, but it does way more things.

Author: how else could you write it. my problem is that the response comes back successfully (201), even if there are problems in it. I would like to make that clearer

Reviewer: wtb `if the Connector states your Request as valid`?

### App: Initiate a Relationship

When the app is opened and no profile has been created yet, the user must create one. From the profile overview, the user can add a new contact using the "Add Contact" option. A QR code must be scanned to complete the process.

> Scanning the qr code should result in a screen similar to the one below, where you can see the information that you added as `content` to the RelationshipTemplate.

Reviewer 3: I think we capitalize QR

> ![Add contact screen](/assets/images/add-contact-screen.jpg){: width="40%"}

Reviewer: you forgot the `| relative_url`

In order to fetch the Message, we need to synchronize the Connector with the Backbone again.

{% include rapidoc api_route_regex="^post /api/v2/Account/Sync$" %}

After syncing, all messages can be displayed with the `GET /api/v2/Messages` route.

> {% include rapidoc api_route_regex="^post /api/v2/Messages$" %}

Reviewer 3: ```suggestion
{% include rapidoc api_route_regex="^get /api/v2/Messages$" %}
```

In order to fetch the Message, we need to synchronize the Connector with the Backbone again.

{% include rapidoc api_route_regex="^post /api/v2/Account/Sync$" %}

After syncing, all Messages can be displayed with the `GET /api/v2/Messages` route.

{% include rapidoc api_route_regex="^get /api/v2/Messages$" %}

> The response should contain a Message with the `content` you entered in the App.

Reviewer: we should also talk about that there will be an event that you can subscribe to. Take the entry to events as soon and as often as possible
